File: backend/app/schemas/statement.py
from datetime import date
from uuid import UUID

from pydantic import BaseModel

from app.schemas.ai_analysis import AiAnalysisPublic
from app.schemas.common import OrmModel
from app.schemas.political_party import PoliticalPartyPublic


class StatementPolitician(OrmModel):
    id: UUID
    slug: str
    full_name: str


class MediaAssetPublic(BaseModel):
    id: UUID
    file_path: str
    media_type: str
    original_filename: str | None = None


class StatementListItem(BaseModel):
    id: UUID
    title: str | None = None
    source_type: str
    statement_date: date | None = None
    politician: StatementPolitician
    party_at_statement_time: PoliticalPartyPublic | None = None
    overall_score: int | None = None


class StatementPublic(StatementListItem):
    source_url: str | None = None
    original_text: str
    media: list[MediaAssetPublic]
    ai_analysis: AiAnalysisPublic | None = None

File: frontend/src/types/statement.ts
import type { AiAnalysis } from "./aiAnalysis";
import type { PoliticalParty } from "./politicalParty";

export type StatementPolitician = {
  id: string;
  slug: string;
  full_name: string;
};

export type MediaAsset = {
  id: string;
  file_path: string;
  media_type: string;
  original_filename?: string | null;
};

export type StatementListItem = {
  id: string;
  title?: string | null;
  source_type: string;
  statement_date?: string | null;
  politician: StatementPolitician;
  party_at_statement_time?: PoliticalParty | null;
  overall_score?: number | null;
};

export type Statement = StatementListItem & {
  source_url?: string | null;
  original_text: string;
  media: MediaAsset[];
  ai_analysis?: AiAnalysis | null;
};
